scipy/interpolate/interpolate.py

```diff
@@ -336,10 +336,11 @@ class interp1d(_Interpolator1D):
         a value outside of the range of x (where extrapolation is
         necessary). If False, out of bounds values are assigned `fill_value`.
         By default, an error is raised.
-    fill_value : float, optional
+    fill_value : float or str, optional
         If provided, then this value will be used to fill in for requested
         points outside of the data range. If not provided, then the default
-        is NaN.
+        is NaN. If set to 'extrapolate' then points outside the data range
+        will be extrapolated.
```

Comment:
If it doesn't work, it shouldn't be allowed. However, I'm not sure if
the different methods work or not --- this is something that should be
checked in this PR.